test/test_cuda.py

```diff
@@ -964,6 +964,48 @@ def test_huge_index(self):
         res_cpu = src.cpu()[idx.cpu()]
         self.assertEqual(res.cpu(), res_cpu)
 
+    def test_randint_randomness_for_large_range(self) -> None:
+        # For large ranges, randint generation is slightly different. This lead to a subtle bug where some Philox
+        # offsets were not calculated correctly, resulting in reused random states.
+        # See https://github.com/pytorch/pytorch/issues/125224
+        size = 1_000_000
+        high = 6_000_000_000  # Keep this above 2**32
+
+        def run(dev: torch.device) -> int:
+            # Measure how many unique numbers are generated in 2 consecutive calls to randint. If random states are
+            # reused, this will yield fewer unique numbers.
+            gen = torch.Generator(device=dev)
+            gen.manual_seed(0)
+            ls = []
+            ls.extend(
+                torch.randint(
+                    0,
+                    high,
+                    torch.Size([1, size]),
```

Comment:
Out of curiosity, what's the need for generate a `[1, size]` tensor and then immediately `.view(-1)` it? Is it to meet some criteria for the unrolling?

Reply:
If I recall correctly, this was necessary to trigger the relevant code path, but I will rewrite this test anyway as I found an easier way to test the fix.